beacon-chain/sync/data_columns_reconstruct.go

```diff
@@ -23,9 +23,25 @@ func recoverBlobs(
 	columnsCount int,
 	blockRoot [fieldparams.RootLength]byte,
 ) ([]cKzg4844.Blob, error) {
-	recoveredBlobs := make([]cKzg4844.Blob, 0, fieldparams.MaxBlobsPerBlock)
+	// Check if all columns have the same length.
+	var blobCount *int
```

Comment:
why are we setting this as a pointer ?

Reply:
Because golang does not implement optionals.
If `blobCount` is nil, we later consider that it is not (yet) set:

```go
		if blobCount == nil {
			blobCount = &length
			continue
		}
```

Reply:
In Python, I would do something like:
```python
blob_count = int | None
...
if blob_count is None:
    blob_count = length
    continue

if blob_count != length:
    raise ColumnsSizeDifferError()
```

Reply:
Why not just set the blobcount before hand ? 
```
blobCount := len(dataColumnSideCars[0].DataColumn)
```

Reply:
Fixed in 0d71dbc07bfb974d35f02a7c6feef6b40c7c7b91.